beta-src/src/components/controllers/WDMainController.tsx

```diff
@@ -1,38 +1,50 @@
+import { Box } from "@mui/material";
 import * as React from "react";
 import {
-  fetchGameData,
   fetchGameOverview,
   gameApiSliceActions,
   gameOverview,
-  userActivity,
+  gameUserActivity,
+  gameData,
+  gameStatus,
+  loadGameData,
 } from "../../state/game/game-api-slice";
 import { useAppDispatch, useAppSelector } from "../../state/hooks";
 import debounce from "../../utils/debounce";
+import getPhaseKey from "../../utils/state/getPhaseKey";
+import WDGameProgressOverlay from "../ui/WDGameProgressOverlay";
 
 const WDMainController: React.FC = function ({ children }): React.ReactElement {
+  const [displayedPhaseKey, setDisplayedPhaseKey] = React.useState<
+    string | null
+  >(null);
   const dispatch = useAppDispatch();
-  const { season, year, processTime, makeNewCall } =
-    useAppSelector(userActivity);
-  const {
-    season: newSeason,
-    year: newYear,
-    processTime: newProcessTime,
-    gameID,
-    user: {
-      member: { countryID },
-    },
-  } = useAppSelector(gameOverview);
-  if (makeNewCall) {
-    dispatch(fetchGameOverview({ gameID: String(gameID) }));
+  const userActivity = useAppSelector(gameUserActivity);
+  const overview = useAppSelector(gameOverview);
+  const { data } = useAppSelector(gameData);
+  const status = useAppSelector(gameStatus);
+
+  const { countryID } = overview.user.member;
+
+  const overviewKey = getPhaseKey(overview);
+  const statusKey = getPhaseKey(status);
+  const dataKey = data.contextVars
+    ? getPhaseKey(data.contextVars.context)
+    : "<BAD>";
+
+  if (userActivity.makeNewCall) {
+    dispatch(fetchGameOverview({ gameID: String(overview.gameID) }));
   }
-  if (
-    processTime &&
-    (season !== newSeason || year !== newYear || processTime !== newProcessTime)
-  ) {
-    dispatch(
-      fetchGameData({ gameID: String(gameID), countryID: String(countryID) }),
-    );
+  const activity = useAppSelector(gameUserActivity);
+  const isPregame = ["", "Pre-game"].includes(overview.phase);
+  const consistentPhase =
+    isPregame || (overviewKey === statusKey && overviewKey === dataKey);
```

Comment:
Question: if isPreGame is true, but also we've managed to load the game data or game status (but not the overview), do bad things happen as we then set consistentPhase true and proceed to try to render that game data or status despite the overview having a missing phase?

Or is there a later check that stops that? Or is this situation not possible for some reason?

Reply:
I don't think that the game data exists when you're in pre-game. But this wasn't well-researched, more an empirical finding.